presto-spi/src/main/java/com/facebook/presto/spi/security/ReadOnlyHttpServletRequest.java

```diff
@@ -0,0 +1,227 @@
+    public Cookie[] getCookies()
+    {
+        return super.getCookies();
+    }
+
+    @Override
+    public long getDateHeader(String name)
+    {
+        return super.getDateHeader(name);
+    }
+
+    @Override
+    public String getHeader(String name)
+    {
+        return super.getHeader(name);
+    }
+
+    @Override
+    public Enumeration<String> getHeaders(String name)
+    {
+        return super.getHeaders(name);
+    }
+
+    @Override
+    public Enumeration<String> getHeaderNames()
+    {
+        return super.getHeaderNames();
+    }
+
+    @Override
+    public int getIntHeader(String name)
+    {
+        return super.getIntHeader(name);
+    }
+
+    @Override
+    public String getMethod()
+    {
+        return super.getMethod();
+    }
+
+    @Override
+    public String getPathInfo()
+    {
+        return super.getPathInfo();
+    }
+
+    @Override
+    public String getContextPath()
+    {
+        return super.getContextPath();
+    }
+
+    @Override
+    public String getQueryString()
+    {
+        return super.getQueryString();
+    }
+
+    @Override
+    public String getRemoteUser()
+    {
+        return super.getRemoteUser();
+    }
+
+    @Override
+    public boolean isUserInRole(String role)
+    {
+        return super.isUserInRole(role);
+    }
+
+    @Override
+    public Principal getUserPrincipal()
+    {
+        return super.getUserPrincipal();
+    }
+
+    @Override
+    public String getRequestedSessionId()
+    {
+        return super.getRequestedSessionId();
+    }
+
+    @Override
+    public String getRequestURI()
+    {
+        return super.getRequestURI();
+    }
+
+    @Override
+    public StringBuffer getRequestURL()
+    {
+        return super.getRequestURL();
+    }
+
+    @Override
+    public String getServletPath()
+    {
+        return super.getServletPath();
+    }
+
+    @Override
+    public HttpSession getSession(boolean create)
+    {
+        return super.getSession(create);
+    }
+
+    @Override
+    public HttpSession getSession()
+    {
+        return super.getSession();
+    }
+
+    @Override
+    public String changeSessionId()
+    {
+        return super.changeSessionId();
+    }
+
+    @Override
+    public boolean isRequestedSessionIdValid()
+    {
+        return super.isRequestedSessionIdValid();
+    }
+
+    @Override
+    public boolean isRequestedSessionIdFromCookie()
+    {
+        return super.isRequestedSessionIdFromCookie();
+    }
+
+    @Override
+    public boolean isRequestedSessionIdFromURL()
+    {
+        return super.isRequestedSessionIdFromURL();
+    }
+
+    @Override
+    public boolean isRequestedSessionIdFromUrl()
+    {
+        return super.isRequestedSessionIdFromUrl();
+    }
+
+    @Override
+    public Map<String, String[]> getParameterMap()
+    {
+        return super.getParameterMap();
+    }
+
+    @Override
+    public String[] getParameterValues(String name)
+    {
+        return super.getParameterValues(name);
+    }
```

Comment:
```suggestion
```

Just remove these, they will all use the parent's implementation if they are not specified.